vue-app/src/views/FactoryRound.vue

```diff
@@ -0,0 +1,771 @@
+        />
+        <router-view :key="$route.path" />
+      </div>
+      <div
+        id="cart"
+        v-if="isSideCartShown && !isHistoricRound"
+        :class="`desktop ${isCartToggledOpen ? 'open-cart' : 'closed-cart'}`"
+      >
+        <cart-widget />
+      </div>
+    </div>
+    <mobile-tabs v-if="isMobileTabsShown" />
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Watch } from 'vue-property-decorator'
+
+import { getCurrentRound } from '@/api/round'
+import { User } from '@/api/user'
+
+import RoundInformation from '@/views/RoundInformation.vue'
+import NavBar from '@/components/NavBar.vue'
+import CartWidget from '@/components/CartWidget.vue'
+import MobileTabs from '@/components/MobileTabs.vue'
+import BackLink from '@/components/BackLink.vue'
+
+import {
+  LOAD_USER_INFO,
+  LOAD_ROUND_INFO,
+  LOAD_RECIPIENT_REGISTRY_INFO,
+  SELECT_ROUND,
+  LOAD_CART,
+  LOAD_COMMITTED_CART,
+  LOAD_CONTRIBUTOR_DATA,
+  LOGIN_USER,
+} from '@/store/action-types'
+import {
+  SET_CURRENT_USER,
+  SET_ACTIVE_ROUND_ADDRESS,
+  TOGGLE_SHOW_CART_PANEL,
+} from '@/store/mutation-types'
+
+@Component({
+  name: 'clr.fund',
+  metaInfo: {
+    title: 'clr.fund',
+    titleTemplate: 'clr.fund - %s',
+    meta: [
+      {
+        name: 'git-commit',
+        content: process.env.VUE_APP_GIT_COMMIT || '',
+      },
+    ],
+  },
+  components: {
+    RoundInformation,
+    NavBar,
+    MobileTabs,
+    CartWidget,
+    BackLink,
+  },
+})
+export default class FactoryRound extends Vue {
+  pollingInterval
+
+  async created(): Promise<void> {
+    const factoryAddress =
+      this.$route.params.factoryAddress ||
+      process.env.VUE_APP_CLRFUND_FACTORY_ADDRESS
+    const roundIndex = this.$route.params.roundIndex
+
+    if (!factoryAddress) {
+      throw new Error('No factory address found')
+    }
+
+    // Always fetch the current round and store it. We use it to compare with the
+    // current round that the user might select manually and display a warning
+    // message describing that an old round is being displayed.
+    const currentRoundAddress = await getCurrentRound(factoryAddress)
+    await this.$store.commit(SET_ACTIVE_ROUND_ADDRESS, currentRoundAddress)
+
+    // Set the `currentRound` and `factoryAddress`
+    await this.$store.dispatch(SELECT_ROUND, {
+      roundIndex: roundIndex || currentRoundAddress,
+      factoryAddress,
+    })
+
+    // Execute and start polling round information every 1 min
+    this.loadRound()
+    this.pollingInterval = setInterval(this.loadRound, 60 * 1000)
+  }
+
+  beforeDestroy() {
+    if (this.pollingInterval) {
+      clearInterval(this.pollingInterval)
+    }
+  }
+
+  @Watch('$route.params')
+  async loadAppData() {
+    const { currentRoundAddress, activeRoundAddress, currentFactoryAddress } =
+      this.$store.state
+    const roundIndex = this.$route.params.roundIndex
+    const selectedRound = roundIndex || activeRoundAddress
```

Comment:
Should this not fallback to `currentRoundAddress` vs `activeRoundAddress `? Perhaps I'm getting confused on the terminology.

Behavior should be: if there's no round info in the URL, set round to the latest round (i.e. `FundingRoundFactory.getCurrentRound()`).

Reply:
You want to default to the `activeRound` if there is no route param, not the `currentRound`. Why? because the `currentRound` might be pointing to a previous round, we want to make sure that if the `roundIndex` is not defined, we load the active round.

See this example:
1. User is in an old round `/round/1`
	- `roundIndex`: 1
	- `currentRound`: 1
	- `activeRound`: 2
2. User now wants to see the latest round, he goes to `/round`
	- `roundIndex`: undefined
	- `currentRound`: 
		- while loading this will still be 1
		- once the load ends, this should be 2 (the active round)
	- `activeRound`: 2

In the transition of these two rounds, **when you are loading** the second route, `currentRound` will still be the previous one (`currentRound`: 1) and that is not correct, we want to display the round 2.

Let me know if I confused you even more or if I'm wrong in my thinking.

Reply:
I think we're on the same page but perhaps I'm confused with the terminology.

Is the `activeRound` what's returned by `FundingRoundFactory.getCurrentRound()`?

Reply:
If so, I think I lean towards keeping that terminology consistent, since the contract refers to that `activeRound` as the `currentRound`. Perhaps `selectedRound` would be a clearer term for the given round a user may be viewing.

Reply:
Yes, I totally agree with that thinking.

1) yes, I have to accept doing `activeRound` = `FundingRoundFactory.getCurrentRound()`. That is what we are doing under the hood.
2) I kept the name `currentRound` instead of `selectedRound` because of the refactor I would have needed to do.

I'm not against it. If you consider this to be something important and worth it, I can rework that and refactor the names. Of course it will help the DX of future contributors.